Diplom_3

MariF24 opened the pull request.

MariF24 committed:
git add -f .\target\allure-results\.
git commit -m "Diplom_3 and add allure report"

MariF24 committed:
git add -f .\target\allure-results\.
git commit -m "add allure report"

AlexKitIt commented on locators.py:
```diff
@@ -0,0 +1,142 @@
+    CLOSE_BUTTON_POPUP_ID_ORDER = (By.XPATH, "//button[contains(@class,'Modal_modal__close__TnseK')]")
+
+    #Количество ингредиента
+    COUNT_INGREDIENT = (By.XPATH, "//section[1]/div[2]/ul[1]/a[1]/div[1]/p")
```
Нужно исправить: здесь и далее: использование порядкового номера элемента в dom является весьма нестабильным способом поиска, необходимо привязываться к атрибутам конкретного элемента
Кроме того использовать абсолютные пути нельзя - очень ненадежно, добавится на пути новый элемент - тест ляжет. Все подобные локаторы необходимо переписать

AlexKitIt commented on pages/base_page.py:
```diff
@@ -0,0 +1,113 @@
+        try:
+            return WebDriverWait(self.driver, timeout).until(expected_conditions.presence_of_element_located(locator))
+        except TimeoutException:
+            print(f"Element with locator {locator} not found within {timeout} seconds.")
```
Нужно исправить: принтов быть не должно 
Если нужно логирование, оно делается по другому

AlexKitIt commented on pages/base_page.py:
```diff
@@ -0,0 +1,113 @@
+        else:
+            print(f"Failed to enter text in element with locator {locator}.")
+
+    @allure.step('Проверяем видимость элемента на странице')
+    def wait_for_element_visible(self, locator:tuple, timeout: int = 10):
+        try:
+            return WebDriverWait(self.driver, timeout).until(expected_conditions.visibility_of_element_located(locator))
+        except TimeoutException:
+            print(f"Element with locator {locator} not visible after {timeout} seconds.")
+            return None
+
+
+    @allure.step('Проверяем видимость элемента на странице с регулируемым таймаутом')
+    def wait_for_element_visible_with_adjustable_timeout(self, locator:tuple, timeout: int):
+        try:
+            return WebDriverWait(self.driver, timeout).until(expected_conditions.visibility_of_element_located(locator))
+        except TimeoutException:
+            print(f"Element with locator {locator} not visible after {timeout} seconds.")
+            return None
```
Нужно исправить:здесь и далее:  копипаста методов, кроме того и  первом варианте можно передавать таймаут

AlexKitIt commented on pages/feed_page.py:
```diff
@@ -0,0 +1,129 @@
+
+    @allure.step('Получать URL текущей страницы')
+    def get_current_url(self, url):
+        self.driver.get(url)
```
Нужно исправить: здесь и далее по проекту: все прямые обращения к driver необходимо вынести в base, там описать методы взаимодействия с ними и оттуда за счет наследования классов страниц от base переиспользовать

MariF24 committed:
git add -f .\target\allure-results\.
git commit -m "add allure report"
git commit -m "add Diplom_3_v.2"

AlexKitIt commented on pages/main_page.py:
```diff
@@ -15,11 +15,10 @@ def __init__(self, driver):
-    def get_current_url(self, url):
-        self.driver.get(url)
-        return self.driver.current_url
+    @allure.step('Получить URL страницы Лента заказов')
+    def get_current_url_feed(self, url):
+        self.get_current_url(url)
+        return url
```
Нужно исправить: зачем эта копипаста если url все равно передается. Такой метод необходимо описать в base

MariF24 committed:
git add -f .\target\allure-results\.
git commit -m "add allure report"
git commit -m "add Diplom_3_v.2"
git commit -m "add Diplom_3_v.3"
